BootloaderCorePkg/Stage1B/Stage1B.c

```diff
@@ -125,6 +125,35 @@ PrepareStage2 (
   return Dst;
 }
 
+/**
+  Check digest for external hash store component.
+
+  @param[in] Stage1bParam   Stage1B parameters.
+  @param[in] Src            Extrenal hash store address
+  @param[in] Length         Used length of external hash store
+
+**/
+VOID
+CheckExtHashStoreMbDigestExtend (
+    IN STAGE1B_PARAM   *Stage1bParam,
+    IN UINT8           *Src,
+    IN UINT32           Length
+  )
+{
+  // Calculate External hash srore Digest if PcdMeasuredBootHashMask differs with PcdCompSignHashAlg
+  if (GetCryptoHashAlg(PcdGet32(PcdMeasuredBootHashMask)) != PcdGet8(PcdCompSignHashAlg)){
+    CalculateHash ((UINT8 *)Src, Length, GetCryptoHashAlg(PcdGet32(PcdMeasuredBootHashMask)),

```

Comment:
If the required hash algo is not supported,  the CalculateHash will fail.  In this case, you cannot set the valid bit.

Reply:
Yes, I will update the condition.